vkrunner/vr-script.c

```diff
@@ -1923,108 +1958,181 @@ process_token_replacements(struct load_state *data)
         return true;
 }
 
-static struct vr_script *
-load_script_from_stream(const struct vr_config *config,
-                        const char *filename,
-                        FILE *f)
+static struct load_state *
+load_script_begin(const struct vr_config *config,
+                  const char *filename)
 {
-        struct load_state data = {
-                .config = config,
-                .filename = filename,
-                .line_num = 1,
-                .script = vr_calloc(sizeof (struct vr_script)),
-                .line = VR_BUFFER_STATIC_INIT,
-                .buffer = VR_BUFFER_STATIC_INIT,
-                .commands = VR_BUFFER_STATIC_INIT,
-                .pipeline_keys = VR_BUFFER_STATIC_INIT,
-                .extensions = VR_BUFFER_STATIC_INIT,
-                .buffers = VR_BUFFER_STATIC_INIT,
-                .current_stage = -1,
-                .current_section = SECTION_NONE,
-                .clear_depth = 1.0f
-        };
-        bool res = true;
         int stage;
+        struct load_state *data = vr_calloc(sizeof (struct load_state));
+        assert(data && "Allocating |data| has a problem.");
+
+        assert(config && "|config| must be a valid pointer.");
+        assert(filename && "|filename| must be a valid pointer.");
 
-        vr_pipeline_key_init(&data.current_key);
+        data->config = config;
+        data->filename = filename;
+        data->line_num = 1;
+        data->script = vr_calloc(sizeof (struct vr_script));
+        data->current_stage = -1;
+        data->current_section = SECTION_NONE;
+        data->clear_depth = 1.0f;
```

Comment:
This is missing the initialisers for the vr_buffers. In practice I guess it will just work because the initialiser is all zeros anyway. But for the sake of consistency it would be good to call vr_buffer_init on them.

Reply:
Done.